FAIL: Test_MultipleContainersInTheNewNetwork

**Describe the bug**
Failing unit test.

```
=== RUN   Test_MultipleContainersInTheNewNetwork
2021/08/04 11:03:06 Starting container id: 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Waiting for container id 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Container is ready id: 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Starting container id: b4bfa770dce2 image: postgres:12
2021/08/04 11:03:07 Waiting for container id b4bfa770dce2 image: postgres:12
2021/08/04 11:03:08 Container is ready id: b4bfa770dce2 image: postgres:12
2021/08/04 11:03:08 Starting container id: 85c4278deffc image: rabbitmq:management-alpine
2021/08/04 11:03:08 Waiting for container id 85c4278deffc image: rabbitmq:management-alpine
    network_test.go:99: failed to start container: /bin/sh command not executable
--- FAIL: Test_MultipleContainersInTheNewNetwork (3.49s)

FAIL

Process finished with the exit code 1
```

**Expected behavior**
Test should pass.

**Additional context**
Error is due to a deprecation of the latest rabbitmq:

```
testcontainers-go$ docker logs d5d0643585eb
error: RABBITMQ_DEFAULT_PASS is set but deprecated
error: RABBITMQ_DEFAULT_USER is set but deprecated
error: RABBITMQ_ERLANG_COOKIE is set but deprecated
error: deprecated environment variables detected

Please use a configuration file instead; visit https://www.rabbitmq.com/configure.html to learn more
```

--- a/network_test.go
+++ b/network_test.go
@@ -84,7 +84,7 @@ func Test_MultipleContainersInTheNewNetwork(t *testing.T) {
 	hp := wait.ForListeningPort("5672/tcp")
 	hp.WithStartupTimeout(3 * time.Minute)
 	amqpRequest := ContainerRequest{
-		Image:        "rabbitmq:management-alpine",
+		Image:        "rabbitmq:3.8.19-management-alpine",
 		ExposedPorts: []string{"15672/tcp", "5672/tcp"},
 		Env:          env,
 		AutoRemove:   true,

--- a/wait/http.go
+++ b/wait/http.go
@@ -7,6 +7,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"net"
 	"net/http"
 	"strconv"
@@ -188,7 +189,7 @@ func (ws *HTTPStrategy) WaitUntilReady(ctx context.Context, target StrategyTarge
 	// cache the body into a byte-slice so that it can be iterated over multiple times
 	var body []byte
 	if ws.Body != nil {
-		body, err = io.ReadAll(ws.Body)
+		body, err = ioutil.ReadAll(ws.Body)
 		if err != nil {
 			return
 		}

--- a/wait/http.go
+++ b/wait/http.go
@@ -1,6 +1,7 @@
 package wait
 
 import (
+	"bytes"
 	"context"
 	"crypto/tls"
 	"errors"
@@ -184,12 +185,21 @@ func (ws *HTTPStrategy) WaitUntilReady(ctx context.Context, target StrategyTarge
 	address := net.JoinHostPort(ipAddress, strconv.Itoa(port.Int()))
 	endpoint := fmt.Sprintf("%s://%s%s", proto, address, ws.Path)
 
+	// cache the body into a byte-slice so that it can be iterated over multiple times
+	var body []byte
+	if ws.Body != nil {
+		body, err = io.ReadAll(ws.Body)
+		if err != nil {
+			return
+		}
+	}
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-time.After(ws.PollInterval):
-			req, err := http.NewRequestWithContext(ctx, ws.Method, endpoint, ws.Body)
+			req, err := http.NewRequestWithContext(ctx, ws.Method, endpoint, bytes.NewReader(body))
 			if err != nil {
 				return err
 			}

--- a/wait/http_test.go
+++ b/wait/http_test.go
@@ -62,6 +62,7 @@ func TestHTTPStrategyWaitUntilReady(t *testing.T) {
 	}
 
 	tlsconfig := &tls.Config{RootCAs: certpool, ServerName: "testcontainer.go.test"}
+	var i int
 	dockerReq := testcontainers.ContainerRequest{
 		FromDockerfile: testcontainers.FromDockerfile{
 			Context: workdir + "/testdata",
@@ -72,6 +73,10 @@ func TestHTTPStrategyWaitUntilReady(t *testing.T) {
 			WithResponseMatcher(func(body io.Reader) bool {
 				data, _ := ioutil.ReadAll(body)
 				return bytes.Equal(data, []byte("pong"))
+			}).
+			WithStatusCodeMatcher(func(status int) bool {
+				i++ // always fail the first try in order to force the polling loop to be re-run
+				return i > 1 && status == 200
 			}).
 			WithMethod(http.MethodPost).WithBody(bytes.NewReader([]byte("ping"))),
 	}

